### PR TITLE
[DEVOPS-435] Docker-compose - Self Signed Certificate Nginx

### PR DESCRIPTION
**Pull request by danielapetito**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
[Docker-compose] Self Signed Certificate Nginx

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)

* **Other information**:
[DEVOPS-435](https://jira.cpqd.com.br/browse/DEVOPS-435) [DOJOT] Ambiente de teste (e outros) deve ter https para acessos ao Kong

### CODE REVIEW

**Review by marianoleonardo — CHANGES_REQUESTED** [review]
Deixei alguns comentários

File: README.md [hunk 1]
```diff
@@ -136,3 +138,100 @@ Place the following at the end of the file, then close and save it.
...
+
+```bash
+sudo docker-compose --file docker-compose-dojot-https-self-signed-certificate.yml up --detach
+```
+
+Open up a browser and visit https://<YOUR IP>. 
+
+You should see an error message telling the connection is not safe.
+
+Now you nedd to click on the warning botton next to your IP address.
```
[the remark this answers is not in the document]
Comment (danielapetito) [reply README.md:183]: corrigido

File: README.md [hunk 2]
```diff
@@ -136,3 +138,100 @@ Place the following at the end of the file, then close and save it.
 
 The above command will run every night at 23:00, renewing the certificate and forcing Nginx to restart
 if the certificate is due for renewal.
+
+#### Secure dojot with self-signed certificate
+
+To get dojot running with https with self-signed certificate, that domain/ip must be accessible where dojot is running and inside the docker network.
```
[the remark this answers is not in the document]
Comment (danielapetito) [reply README.md:144]: corrigido

File: README.md [hunk 3]
```diff
@@ -136,3 +138,100 @@ Place the following at the end of the file, then close and save it.
...
+
+You should see an error message telling the connection is not safe.
+
+Now you nedd to click on the warning botton next to your IP address.
+
+Then click in the option telling the certificate is not valid. 
+
+Enter in details tab and export the certificate adding extension ".crt" .
+
+The next step, it is uploading the certificate (that you exported) in your browser.
```
[the remark this answers is not in the document]
Comment (danielapetito) [reply README.md:189]: corrigido

File: README.md [hunk 4]
```diff
@@ -136,3 +138,100 @@ Place the following at the end of the file, then close and save it.
...
+
+```sh
+docker-compose --profile complete up -d
+```
+
+If you will use iotagent-mqtt to send message with self-signed certificate, it is necessary configure the `iotagent-mqtt-cert-sidecar` environment variable `CERT_SC_CERTS_HOSTNAMES` including the `server IP` and `server name`
+
+```sh
+Example:
+CERT_SC_CERTS_HOSTNAMES: '["http-agent", "${DOJOT_DOMAIN_NAME:-localhost}", "<server IP>", "<server Name>"]'
```
[the remark this answers is not in the document]
Comment (danielapetito) [reply README.md:228]: alterado

File: self-signed-certificate-nginx/docker-compose-dojot-https-self-signed-certificate.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    image: nginx:latest
+    restart: always
+    ports:
+      - "80:80"
+      - "443:443"
+    volumes:
+      - ./nginx-dojot-https-self-signed-certificate.conf:/etc/nginx/conf.d/default.conf
+      - ./dojot-nginx-certificate.crt:/etc/nginx/certificate/nginx-certificate.crt
+      - ./dojot-nginx-certificate.key:/etc/nginx/certificate/nginx-certificate.key
+
```
[the remark this answers is not in the document]
Comment (danielapetito) [reply self-signed-certificate-nginx/docker-compose-dojot-https-self-signed-certificate.yml:28]: corrigido

**Review by gabrielifg — APPROVED** [review]
bom trabalho!!!

**Review by marianoleonardo — APPROVED** [review]

**Pull request merged**